### ISSUE TITLE
Write multiple parquet files in parallel

### ISSUE THREAD

**Issue by gab23r** [pola-rs/polars#6506]
### Problem description

Using polars, is it possible/ in the scope to be able to save multiple parquet files in parallel. 

I am thinking of something ~like~ this
```
df1.sink_parquet(path1, lazy = True)
df2.sink_parquet(path2, lazy = True)

pl.collect_all([df1, df2]) # both dataframe are writing to disks here in parallel

```

**Comment by ritchie46** [comment]
We don't hold the GIL. You could spawn an extra thread.

**Comment by gab23r** [comment]
ok, but then if df1 and df2 share a large commun subplan. I have to wait for df1 and df2 to be collected until I can start to write any of them on disk. Correct ?

**Comment by ritchie46** [comment]
Currently a single lazyframe cannot share a common subplan with a node that is not leading to the same plan, eg. via a join or union.

This might be something we add in the future, but is currently out of scope as I cannot keep up. :')

**Comment by gab23r** [comment]
Ok that's clear. And thank you again for this wonderful package !
